holmes/plugins/toolsets/azure_sql/azure_sql_toolset.py

```diff
@@ -0,0 +1,182 @@
+            tags=[ToolsetTag.CORE],
+            tools=[
+                AnalyzeDatabaseHealthStatus(self),
+                AnalyzeDatabasePerformance(self),
+                AnalyzeDatabaseConnections(self),
+                AnalyzeDatabaseStorage(self),
+                GetTopCPUQueries(self),
+                GetSlowQueries(self),
+                GetTopDataIOQueries(self),
+                GetTopLogIOQueries(self),
+                GetActiveAlerts(self),
+                AnalyzeConnectionFailures(self),
+            ],
+        )
+        self._reload_llm_instructions()
+
+    def prerequisites_callable(self, config: Dict[str, Any]) -> Tuple[bool, str]:
+        if not config:
+            return False, TOOLSET_CONFIG_MISSING_ERROR
+
+        errors = []
+        try:
+            azure_sql_config = AzureSQLConfig(**config)
+
+            # Set up Azure credentials
+            try:
+                credential: Union[ClientSecretCredential, DefaultAzureCredential]
+                if (
+                    azure_sql_config.tenant_id
+                    and azure_sql_config.client_id
+                    and azure_sql_config.client_secret
+                ):
+                    logging.info(
+                        "Using ClientSecretCredential for Azure authentication"
+                    )
+                    credential = ClientSecretCredential(
+                        tenant_id=azure_sql_config.tenant_id,
+                        client_id=azure_sql_config.client_id,
+                        client_secret=azure_sql_config.client_secret,
+                    )
+                else:
+                    logging.info(
+                        "Using DefaultAzureCredential for Azure authentication"
+                    )
+                    credential = DefaultAzureCredential()
+
+                # Test the credential by attempting to get tokens for both required scopes
+                mgmt_token = credential.get_token(
+                    "https://management.azure.com/.default"
+                )
+                if not mgmt_token.token:
+                    raise Exception("Failed to obtain Azure management token")
+
+                # Test SQL database token as well
+                sql_token = credential.get_token(
+                    "https://database.windows.net/.default"
+                )
+                if not sql_token.token:
+                    raise Exception("Failed to obtain Azure SQL database token")
+
+            except Exception as e:
+                message = f"Failed to set up Azure authentication: {str(e)}"
+                logging.error(message)
+                errors.append(message)
+                return False, message
+
+            # Store single database configuration and create API client
+            self._database_config = azure_sql_config.database
+            self._api_client = AzureSQLAPIClient(
```

Comment:
I saw that this could also get username & password but we never pass it and it isn't support in the config. 
It is still needed? can we remove it?